examples/poly-model.html

```diff
@@ -0,0 +1,307 @@
+    model.position.set(10000, 10000, 10000);
+    scene.add(model);
+  });
+
+  // Bind our event handlers
+  window.addEventListener('resize', onWindowResize, false);
+  canvas.addEventListener('click', onClick, false);
+
+  // Kick off the render loop!
+  update();
+}
+
+/**
+ * The render loop, called once per frame. Handles updating
+ * our scene and rendering.
+ */
+function update() {
+  // Clears color from the frame before rendering the camera (arView) or scene.
+  renderer.clearColor();
+
+  // Render the device's camera stream on screen first of all.
+  // It allows to get the right pose synchronized with the right frame.
+  arView.render();
+
+  // Update our camera projection matrix in the event that
+  // the near or far planes have updated
+  camera.updateProjectionMatrix();
+
+  // Update our perspective camera's positioning
+  vrControls.update();
+
+  // Render our three.js virtual scene
+  renderer.clearDepth();
+  renderer.render(scene, camera);
+
+  // Kick off the requestAnimationFrame to call this function
+  // when a new VRDisplay frame is rendered
+  vrDisplay.requestAnimationFrame(update);
+}
+
+/**
+ * On window resize, update the perspective camera's aspect ratio,
+ * and call `updateProjectionMatrix` so that we can get the latest
+ * projection matrix provided from the device
+ */
+function onWindowResize () {
+  camera.aspect = window.innerWidth / window.innerHeight;
+  camera.updateProjectionMatrix();
+  renderer.setSize(window.innerWidth, window.innerHeight);
+}
+
+/**
+ * When clicking on the screen, fire a ray from where the user clicked
+ * on the screen and if a hit is found, place a cube there.
+ */
+function onClick (e) {
+  // Inspect the event object and generate normalize screen coordinates
+  // (between 0 and 1) for the screen position.
+  var x = e.clientX / window.innerWidth;
+  var y = e.clientY / window.innerHeight;
+
+  // Send a ray from the point of click to the real world surface
+  // and attempt to find a hit. `hitTest` returns an array of potential
+  // hits.
+  var hits = vrDisplay.hitTest(x, y);
+
+  if (!model) {
+    console.warn('Model not yet loaded');
+    return;
+  }
+
+  // If a hit is found, just use the first one
+  if (hits && hits.length) {
+    var hit = hits[0];
+
+    // Turn the model matrix from the VRHit into a
+    // THREE.Matrix4 so we can extract the position
+    // elements out so we can position the shadow mesh
+    // to be directly under our model. This is a complicated
+    // way to go about it to illustrate the process, and could
+    // be done by manually extracting the "Y" value from the
+    // hit matrix via `hit.modelMatrix[13]`
+    var matrix = new THREE.Matrix4();
+    var position = new THREE.Vector3();
+    matrix.fromArray(hit.modelMatrix);
+    position.setFromMatrixPosition(matrix);
+
+    // Set our shadow mesh to be at the same Y value
+    // as our hit where we're placing our model
+    shadowMesh.position.y = position.y;
```

Comment:
You could make this more future proof by rotating it based on the plane pose so that it would work on non-horizontal surfaces

Reply:
Good call; I think I might have a misunderstanding of the rotation of the hit matrix, but I'll dig into this